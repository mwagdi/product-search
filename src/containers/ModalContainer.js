import { connect } from 'react-redux';
import { closeModal } from '../store/actions';
import Modal from './../components/Modal';

const mapStateToProps = state => {
    return {
        product: state.modal.product
    }
}

export default connect(mapStateToProps,{ closeModal })(Modal);