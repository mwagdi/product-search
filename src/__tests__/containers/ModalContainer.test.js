import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import ModalContainer from '../../containers/ModalContainer';
import { Provider } from 'react-redux';

describe("Modal Container", () => {
    let wrapper;
    const mockStore = configureStore();
    let store;
    const initialState = {
        product: {
            title: "Product Title",
            additional_image_link: "dummy1.jpg,dummy2.jpg"
        }
    }
    beforeEach(() => {
        store = mockStore(initialState)
        wrapper = shallow(<Provider store={store}><ModalContainer /></Provider>);
        
    })

    test('Renders correctly', () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    })
})