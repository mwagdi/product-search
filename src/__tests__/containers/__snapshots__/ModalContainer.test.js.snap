// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Container Renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {
        "product": Object {
          "additional_image_link": "dummy1.jpg,dummy2.jpg",
          "title": "Product Title",
        },
      },
    }
  }
>
  <Connect(Modal) />
</ContextProvider>
`;
